docs/cmds.md

```diff
@@ -0,0 +1,290 @@
+effects are normally represented by the `Cmd msg` type.
+Unfortunately, Elm does not currently provide a way to 
+inspect `Cmd` values to see which commands they represent.
+There's some extra work required to get around this current limitation,
+which the rest of this page will walk you through.
+
+Hopefully you'll agree that the benefits (listed above) of testing your program in this way
+will be worth the extra effort!
+But be aware of this trade-off when considering whether this type of testing is right for your program.
+ 
+:::
+
+
+## Introducing the example program
+
+The rest of this page will be working with an example program
+that interacts with a fictional API for controlling the lights in your house.
+
+This diagram shows the architecture of the example application.
+The Elm code makes HTTP requests to two different endpoints:
+`/v1/devices` to get a list of all the lights that can be controlled,
+and `/v1/devices/:id` to change the setting of a specific light.
+
+![architecture diagram described in the preceding paragraph](./HomeAutomationExample.svg)
+
+
+## Goal: the ideal test
+
+The test we want to write represents the following scenario:
+
+1. The user loads the main page
+1. The page requests the list of lights
+1. The server responds with the list of lights and their current states
+1. The user clicks "Turn on" for the kitchen lights
+1. We expect the page to send a request to the server with the correct values
+
+Here's what that test will look like in code:
+
+```elm
+import Expect
+import Test exposing (test)
+import TestContext
+
+test "controlling a light" <|
+    \() ->
+        start
+            |> TestContext.simulateHttpOk
+                "GET"
+                "http://localhost:8003/lighting_service/v1/devices"
+                """[{"id":"K001", "name":"Kitchen", "dimmable":false, "value":0}]"""
+            |> TestContext.clickButton "Turn on"
+            |> TestContext.assertHttpRequest
+                "POST"
+                "http://localhost:8003/lighting_service/v1/devices/K001"
+                (.body >> Expect.equal """{"value":1}""")
+            |> TestContext.done
+```
+
+However, running this test produces the following failure:
+
+```
+↓ HomeAutomationExampleTest
+✗ controlling a light
+
+    TEST SETUP ERROR: In order to use simulateHttpResponse,
+    you MUST use TestContext.withSimulatedEffects before calling TestContext.start
+
+
+TEST RUN FAILED
+```
+
+As the error explains, we must use `TestContext.withSimulatedEffects`
+before the test will work.
+But using it requires us to provide a function of type `effect -> TestContext.SimulatedEffect`.
+Because the effect type of our program
+(the type that `init` and `update` return as the second item in the tuple)
+is currently `Cmd Msg` (which Elm does not currently allow us to inspect the values of),
+we'll need to do the following before the test will work:
+
+1. make a new type which can represent all the effects our program can produce
```

Comment:
Thanks for breaking this run on sentence out into a list! It's much more clear. These are all stand alone sentences so they should start with a capital and end with a period